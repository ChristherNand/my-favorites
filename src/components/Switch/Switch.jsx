import React from 'react';
import "./Switch.css"

class Switch  extends React.Component{

    state = {
        active: true
    }

    swap = () => {
        this.setState(prev => ({active: !prev.active}))
    }

    render(){
        console.log(this.state);
        const {icon, name = ""} = this.props;
        let className = this.state.active? "slice": "slice --active";

        return  <div className="switch" onClick={this.swap}>
                    {icon && <i className="material-icons">{icon}</i>}
                    {name}
                    <div className={className}></div>
                </div>
    }
}

export default Switch ;