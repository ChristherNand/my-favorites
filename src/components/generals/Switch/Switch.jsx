import React, {useState} from 'react';
import "./Switch.css"

function Switch(props) {

    const {icon, name = ""} = props;
    const [active, setActive] = useState(true);

    function swap(){
        setActive(!active)
    }

    console.log("active", active);
    let className = active? "slice": "slice --active";

    return  <div className="switch" onClick={swap}>
                {icon && <i className="material-icons">{icon}</i>}
                {name}
                <div className={className}></div>
            </div>
}

export default Switch ;