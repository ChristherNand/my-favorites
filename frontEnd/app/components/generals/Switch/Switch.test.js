import React, {useState} from 'react';
import '../../../setupTests';
import { shallow, mount } from 'enzyme';
import Switch from './Switch'

var wrapper;
beforeEach(() => {
    wrapper = shallow(<Switch/>);
})

describe('component rendering', () =>{    
    
    it('render', () => {
        wrapper = shallow(<Switch/>);
        expect(wrapper.html()).toMatchSnapshot();
    })
})