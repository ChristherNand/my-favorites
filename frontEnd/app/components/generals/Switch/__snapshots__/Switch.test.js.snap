// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component rendering render 1`] = `"<div class=\\"switch\\"><div class=\\"slice\\"></div></div>"`;

exports[`component rendering render 2`] = `"<div class=\\"switch\\"><div class=\\"slice\\"></div></div>"`;
